url8417b1b213_Ratchette_Rummikub_tgz-pJ8-rummikub_RummikubJ8/src/rummikub/Server.java

```diff
@@ -0,0 +1,404 @@
+		}
+		else
+			return Integer.parseInt(args[0]);
+		
+		return -1;
+	}
+
+    
+    
+    // **********************************************************
+    // 						Initialization 
+    // **********************************************************
+    
+    /**
+     * Constructor for the server
+     * @param numPlayers The number of players per game
+     */
+    public Server(int numPlayers){
+    	serverSocket = null;
+        clientSocket = new Socket[numPlayers];
+        inbox =  new BufferedReader[numPlayers];
+        outbox = new PrintWriter[numPlayers];
+        
+        pool = null;
+        hands = null;
+    	game = null;
+    	
+    	printStatus("Initialization complete");
+    }
+    
+    
+    
+    // **********************************************************
+    // 							Daemon
+    // **********************************************************
+    
+	/**
+	 * Runs the server as a thread
+	 * @param args command line arguments sent to main
+	 */
+    public static void main(String[] args) {
+		int numPlayers;
+		
+		numPlayers = validateArguments(args);
+		if(numPlayers == -1){
+			printUsage();
+			return;
+		}
+		
+		(new Server(numPlayers)).start();
+	} 
+    
+    /**
+     * The server's background thread
+     * NOTE: this function should never be called directly, it should be invoked
+     * 		through the start method of java threads.
+     * 
+     *  for more information, please see: 
+     *  	http://docs.oracle.com/javase/tutorial/essential/concurrency/
+     */
+    @Override
+    public void run(){
+        String message;
+    	
+        acceptClients();
+        startGame();
+                
+        while(!game.isGameOver()){
+        	updateTurn();
+        	printStatus("It is now " + GameInfo.indexToPlayerName(this.turn) + "'s turn");
+        	printStatus("Tiles in pool: " + pool.numTilesRemaining());
+        	
+        	// broadcast the current state of the board to next player
+        	System.out.println(game.displayGame());
+        	outbox[turn].println(game.toString());
+        	
+        	// receive the player's move
+            try{
+                message = inbox[turn].readLine();
+                interpretMessage(message);
+            }
+            
+            catch(Exception e){
+            	// FIXME Analyze the exception thrown
+                printStatus("A client has disconnected. Relaying game termination to all clients");
+                this.turn = GameInfo.DISCONNECT;
+                break;
+            }
+        }
+        
+        if(game.isGameOver())
+        	broadcastFinalScores();
+        
+        this.disconnectClients();
+    }
+    
+    /**
+     * Prints the current status of the server
+     * This function should be called at least once from every other function in the server
+     * @param message	What is going on
+     */
+	private void printStatus(String message){
+		String date = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.